homeassistant/components/insteon/__init__.py

```diff
@@ -24,21 +25,71 @@
     CONF_SUBCAT,
     CONF_UNITCODE,
     CONF_X10,
-    CONF_X10_ALL_LIGHTS_OFF,
-    CONF_X10_ALL_LIGHTS_ON,
-    CONF_X10_ALL_UNITS_OFF,
     DOMAIN,
-    INSTEON_ENTITIES,
+    INSTEON_COMPONENTS,
+    ON_OFF_EVENTS,
 )
 from .schemas import CONFIG_SCHEMA  # noqa F440
-from .utils import async_register_services, register_new_device_callback
+from .utils import (
+    add_on_off_event_device,
+    async_register_services,
+    get_device_platforms,
+    register_new_device_callback,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 
+async def async_id_unknown_devices(config_dir):
+    """Send device ID commands to all unidentified devices."""
+    _LOGGER.info("Identifying Insteon devices")
+    await devices.async_load(id_devices=1)
+    for addr in devices:
+        device = devices[addr]
+        flags = True
+        for name in device.operating_flags:
+            if not device.operating_flags[name].is_loaded:
+                flags = False
+                break
+        if flags:
+            for name in device.properties:
+                if not device.properties[name].is_loaded:
+                    flags = False
+                    break
+        if not device.aldb.is_loaded or not flags:
+            await device.async_read_config()
```

Comment:
Does this need to be awaited sequentially or can we use `asyncio.gather` to await all the tasks of the loop concurrently?